### PR TITLE
Fix vector draped polyline widths at large widths

### PR DESCRIPTION
**Pull request by danielzhong**
# Description
Before:
image

After:
image

## World space (`widthUnits: "meters"`)

Follow-up to #13703. Two issues affecting wide strokes in world space units:

1. The intersection test between vector polyline collections and 3D/Globe tiles checks each tile's rectangle against a rectangle bounding all polyline vertices in the collection. Even if the polyline's vertices don't intersect a given tile, a polyline with a wide enough stroke should still paint it. A correct render would include that tile, but the current intersection test excludes it, and so the polyline stroke is cut off along the tile's edge. The problem gets worse when zooming in: the polyline stroke takes up more screen space (since its width is fixed in meters), so more tiles are affected. To remediate this, we pad the polyline collection's rectangle by half a stroke-width (using the maximum stroke across all polylines in the collection).

2. Converting a stroke's width from meters into the tile's UV space (for shading) requires knowing how many meters one unit of that UV covers. Each tile computes this via a flat plane that is tangent at the tile's center, which only approximates the globe's curvature, and increases in error toward the tile's edges. Since neighboring tiles have different tangent planes, strokes become discontinuous at tile boundaries. This effect is more visible the wider the stroke width in meters. To remediate this, widths in meters are now clamped to MAXIMUM_GROUND_WIDTH_ANGLE (0.1 radians of ground angle, about 640 km), with a one-time warning for exceeding this limit.

## Screen space (`widthUnits: "pixels"`)
https://github.com/CesiumGS/cesium/pull/13737/commits/ae5276d65cfa27613d2aaa27add18516c4755a49
Wide strokes shredded on steep terrain and ballooned into blobs when zooming out.

Previously, we took screen-space derivatives over a polyline fragment's UV coordinates to approximate how much a polyline's UV changed per pixel. Given this, we could convert a fragment's UV-distance to the nearest point on the polyline centerline into a pixel-distance. However, this approach has an issue: at certain camera angles, pixels can span large distances. The screen space derivatives blow up, and we end up painting fragments that are actually outside the stroke width distance. The error here gets worse as the polyline width grows.

To address this, we measure distance to the polyline in meters first (as with world-space polylines). Then, we convert a polyline's pixel-width into meters using `czm_metersPerPixel`, which only needs a fragment's depth (and camera state) to work. No screen space derivatives needed. No error introduced by grazing camera angles. This is what `GroundPolylinePrimitive` and 3D Tiles clamped polylines do already, and it simplifies the shader logic quite a bit as well.

____

One more thing this turned up: dirty regions are cleared every frame, but only tiles rendered *that* frame were re-baked, so a collection changed while a tile was off screen left that tile with stale data. You'd see the old width when zooming back out. Every loaded tile is now checked and released, then rebuilt lazily when it's rendered, the same way clipping polygon data works.

## Testing plan
[sandcastle](https://ci-builds.cesium.com/cesium/DanielZ/feat/vector-line-large-width/Apps/Sandcastle2/index.html#c=lVVtb9s2EP4rBxUb7EFWZCcNMikJ1rjYWqBugqTtPkxDQUsn+xCKFEjKdmL4vw+kKMtOsDfAsHmvfO6545mqWioDPwHTMEVNTQWlkhVkQe6kLEgzlYlcCm0gZ8qgJiY0XHnvaNrpTiMb+B4XClG/U4o9Df4YjceTaDIO4fQ8in8OwcnjcyePx508ae0XXraHQ3t86uP/HB5gqaUmQ9JBEbiGX7lk5vysvfg1tprlj62tL2J4mI8EGWL8G+EarmALBWpDgtkrkn+udXBUxSSEs7dxPAxBKkJhfIYtLJEVJBYJxFEcQk0mX+4Tz5hZRkbes8LiGoxO48hmUJJz5w872DmsZSNymxByhcygRTugIoQ1FWb5VZDR/jyErfUHaKtbEa5Rear8pd+czoVvwaBSjPpSv7Syq/N3qXjhFYNhCHOm8RN7QnVH+SOqBErGNYagcxQ4kwW+0JcN5zpXiOKmMcay4Q1LWeELlWArWjjOPiCvXxgXKHNZHGRmgipPsNcYqpCTwL2CRClv5KYHiRwdhR9FQTkzsssGu3YcOspcNXDlqYuceOSQS+5THfN605QlqjvJnyyQ6d5r4DtSK6rI0AqnshFmxjYJjMPWtEJlcNPr+1mNOIqFWXq/JdJiae6xRIUix33XPhzro+mnd7O7719uv/92f/v183sf3Q9Lcjg41njAQQc7YkUx2PbvLemPIVTMoCLGk78nYOZdBlub1NLdvSYrRdP7j7OH289+bD0g2A37djjmoz1r2gHq8XVuvk85q1CxSKNpH0f/rDtHhaZRwvs73c6tgf5FZcHajrw9Z0EIWVChQaXteRzH9m2mryLaAe9Datog70JsRBAGl9o8cbz+hdql2yg+iKITg1XNmUF9Mm/yRzRRrvUwtUjftDt4KoVhJFDZtUS65uwpgZLjJu1IG8fxD6mfik7a2QyRrRL67iWgkDNLY9qnImH7NJpzmT/+x5QjzubIjxKzuZa8MZiCkXUCb+tNChxL44/PIxIF2llPoWZFuwrP6g1cWOtcqgLVSLGCGu30KcxZ/rhQshFFAmoxZ4OzSdh+4uhimHbT9AaLsizLFEopzEjTMyYwntQmhVqSMKhGuEJhZ11Iga6Iy5O2EZm4LGgFVFx1f3d7qrMAcs60vsoCu74e6Bmz4NrW70I628r1+7pPczA4r5xOClpdvwpvqcyCa7dmQdcsR/ixKphepjCJY6h8oPv+dwSHg/h/ITy42BcYzqHeHENof/8C)

## Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [x] I have updated `CHANGES.md` with a short summary of my change
- [x] I have added or updated unit tests to ensure consistent code coverage
- [x] I have updated the inline documentation, and included code examples where relevant
- [x] I have performed a self-review of my code

## AI acknowledgment

- [x] I used AI to generate content in this PR
- [x] If yes, I have reviewed the AI-generated content before submitting

If yes, I used the following Tools(s) and/or Service(s):
Github Copilot

If yes, I used the following Model(s):
Claude Opus 5

### CODE REVIEW

File: packages/engine/Source/Scene/GlobeSurfaceTileProvider.js [hunk 1]
```diff
@@ -419,21 +419,25 @@ class GlobeSurfaceTileProvider {
     const vectorProvider = this._vectorProvider;
     vectorProvider.minimumTileScreenPixels = minimumTileScreenPixels(this);
     vectorProvider.update(frameState.frameNumber);
-    this._quadtree.forEachRenderedTile(
+    this._quadtree.forEachLoadedTile(
```
Comment (mzschwartz5) [inline packages/engine/Source/Scene/GlobeSurfaceTileProvider.js:422]: I could swear we had this issue before and we've gone back and forth between `forEachRenderedTile` and `forEachLoadedTile`.

That first change had to do with performance, IIRC? @donmccurdy you have more context, I believe. Did it have something to do with constantly rebaking the root tiles even when not needed, or was that a red herring?

File: packages/engine/Source/Shaders/VectorCommon.glsl [hunk 2]
```diff
@@ -9,11 +9,9 @@ uniform highp sampler2D u_vectorSegmentTexture;
 uniform highp sampler2D u_vectorWidthTexture;
 uniform highp sampler2D u_vectorSegmentPrimitiveIndicesTexture;
 uniform highp sampler2D u_vectorGridCellIndicesTexture;
-#ifdef VECTOR_WIDTH_IN_METERS
```
Comment (mzschwartz5) [inline packages/engine/Source/Shaders/VectorCommon.glsl:12]: How come we're able to remove this macro now?

File: packages/engine/Source/Core/VectorPipeline.js [hunk 3]
```diff
@@ -29,6 +30,9 @@ import Rectangle from "./Rectangle.js";
 const GRID_TARGET_SEGMENTS_PER_CELL = 16;
 const GRID_NEIGHBOR_PADDING_SCALE = 0.35;
 
+// A tile measures distances on a plane tangent at its center, so a stroke spanning more ground angle than this steps at tile boundaries.
+const MAXIMUM_GROUND_WIDTH_ANGLE = 0.1;
```
Comment (mzschwartz5) [inline packages/engine/Source/Core/VectorPipeline.js]: What do you mean by "steps at tile boundaries"? (Is that supposed to say "stops"?)

File: packages/engine/Source/Shaders/VectorCommon.glsl (hunk 2 above)
Comment (danielzhong) [reply packages/engine/Source/Shaders/VectorCommon.glsl:12]: Every width mode now measures distance on the ground first, so `u_vectorMetersPerUv` is needed by all of them, not just meter widths.

File: packages/engine/Source/Scene/GlobeSurfaceTileProvider.js (hunk 1 above)
Comment (danielzhong) [reply packages/engine/Source/Scene/GlobeSurfaceTileProvider.js:422]: I think it was f4b2dca9 (#13577), and the perf concern was about baking, not invalidation. That commit fused "test dirty + release + rebake" into a single `updateTileData` call, and a fused call can only run where baking is allowed, so invalidation ended up confined to rendered tiles too. All I've done here is pull them back apart, `isTileDirty` is just a predicate now, so releasing can happen on loaded tiles while baking still stays on rendered. Basically the same shape as clipping, which dodged this by never merging them in the first place.

File: packages/engine/Source/Core/VectorPipeline.js (hunk 3 above)
Comment (mzschwartz5) [reply packages/engine/Source/Core/VectorPipeline.js]: Also, reminder for change log entry. I'm not sure if this constitutes a breaking change, but we should at least indicate the new limitation. This should be in some public facing documentation as well.

File: packages/engine/Source/Core/VectorPipeline.js [hunk 4]
```diff
@@ -173,20 +186,46 @@ class VectorPipeline {
       _writePickColor(pickColors, i, polyline._pickId);
     }
 
+    if (isWidthClamped) {
+      oneTimeWarning(
+        "vector-ground-width-clamped",
+        `Polyline widths in meters are clamped to ${maximumGroundWidth.toFixed(0)} meters. Wider strokes step at tile boundaries.`,
```
Comment (mzschwartz5) [inline packages/engine/Source/Core/VectorPipeline.js]: (Again, not sure what is meant by "Wider strokes 'step' at tile boundaries")

File: packages/engine/Source/Core/VectorProvider.js [hunk 5]
```diff
@@ -628,19 +653,65 @@ function computeMetersPerUv(rectangle, ellipsoid) {
   );
 }
 
+/**
+ * The ground a collection covers: the rectangle its geometry occupies, grown by
+ * half the width of its widest line. The margin is a fraction of the target
+ * rectangle, matching the one segments are clipped with, so a target admitted
+ * here is one that can pack content.
```
Comment (mzschwartz5) [inline packages/engine/Source/Core/VectorProvider.js]: > ... The margin is a fraction of the target rectangle, matching the one segments are clipped with, so a target admitted here is one that can pack content.

I'm not sure what this means. Is there a way to reword it that is clearer? (E.g. what is a "target" here and what are we "admitting" it to and what is the criterion for admission. These things are all a little vague in my opinion.)

File: packages/engine/Source/Core/VectorProvider.js [hunk 6]
```diff
@@ -628,19 +653,65 @@ function computeMetersPerUv(rectangle, ellipsoid) {
...
+  result.south -= uvMargin * rectangle.height;
+  result.north += uvMargin * rectangle.height;
+
+  if (collectionRectangle.width + 2.0 * marginLongitude >= CesiumMath.TWO_PI) {
+    result.west = -CesiumMath.PI;
+    result.east = CesiumMath.PI;
+  } else {
+    result.west = CesiumMath.negativePiToPi(result.west - marginLongitude);
+    result.east = CesiumMath.negativePiToPi(result.east + marginLongitude);
+  }
```
Comment (mzschwartz5) [inline packages/engine/Source/Core/VectorProvider.js:689]: What is the special case here? Does `negativePiToPi` not already take care of the first case?

File: packages/engine/Source/Shaders/VectorCommon.glsl [hunk 7]
```diff
@@ -132,14 +119,13 @@ vec4 vectorPolylineRender(vec2 vectorUv, vec4 baseColor)
...
-            : length(screenFromUv * offsetToLine) - halfWidth;
+        float halfWidthMeters = width < 0.0 ? halfWidth : halfWidth * metersPerPixel;
 #elif defined(VECTOR_WIDTH_IN_METERS)
-        float edgeDistance = (length(metersFromUv * offsetToLine) - halfWidth) * pixelsPerMeter;
+        float halfWidthMeters = halfWidth;
 #else
-        float edgeDistance = length(screenFromUv * offsetToLine) - halfWidth;
+        float halfWidthMeters = halfWidth * metersPerPixel;
 #endif
+        float edgeDistance = (length(metersFromUv * offsetToLine) - halfWidthMeters) / metersPerPixel;
```
Comment (mzschwartz5) [inline packages/engine/Source/Shaders/VectorCommon.glsl:128]: Why is `edgeDistance` now always measured in pixels, regardless of `VECTOR_WIDTH_IN_METERS`? Pre-diff, it was _either_ in meters, _or_ in pixels -- path depedendent.

**Review by mzschwartz5 — COMMENTED** [review]
Sandcastle works well, root cause analysis makes sense to me and the proposed solutions do as well. The code changes are still a little hard for me to fully digest and vet. I left some questions (no changes requested), and I may need to come back to this later and think some more.

File: packages/engine/Source/Core/VectorPipeline.js (hunk 3 above)
Comment (danielzhong) [reply packages/engine/Source/Core/VectorPipeline.js]: "step" as in a staircase, not "stop", bad wording on my part. Each tile measures distance on a plane tangent at its own center, so once a stroke spans enough ground angle, neighbouring tiles disagree and its edges stop lining up across the seam. Reworded the comment. Thanks!

File: packages/engine/Source/Core/VectorProvider.js (hunk 5 above)
Comment (danielzhong) [reply packages/engine/Source/Core/VectorProvider.js]: Fair, that sentence was doing too much. "Target" was just the tile rectangle, so I renamed the parameter to `tileRectangle` and dropped the vague framing. The only real point is that the margin has to use the same conversion as the segment clipping at bake time, otherwise we'd mark tiles dirty that bake out empty, or miss tiles that do pick up geometry

File: packages/engine/Source/Core/VectorProvider.js (hunk 6 above)
Comment (danielzhong) [reply packages/engine/Source/Core/VectorProvider.js:689]: negativePiToPi normalizes one angle at a time and doesn't know the two are a pair. A global collection padded by 0.1 gives west = π - 0.1, east = -π + 0.1, which Rectangle reads as a 0.2 rad sliver crossing the antimeridian, the complement of the globe. Those tiles would then never be marked dirty.

File: packages/engine/Source/Shaders/VectorCommon.glsl (hunk 7 above)
Comment (danielzhong) [reply packages/engine/Source/Shaders/VectorCommon.glsl:128]: It was always in pixels. The old meters path ended in `* pixelsPerMeter`. It has to be, since the only consumer is the smoothstep against vectorCoverageRadius, which is half a pixel.

What changed is the ruler, not the unit. There used to be two (screenFromUv and pixelsPerMeter), both built from screen derivatives, so both measured pixels along the draped surface and were orientation-dependent (that's the bug). Now there's one: measure on the ground, divide once by `czm_metersPerPixel`, which only depends on depth.

File: packages/engine/Source/Scene/BufferPolylineCollection.js [hunk 8]
```diff
@@ -75,12 +75,13 @@ class BufferPolylineCollection extends BufferPrimitiveCollection {
...
    * @param {boolean} [options.debugShowBoundingVolume=false]
-   * @param {BlendOption} [options.blendOption=BlendOption.TRANSLUCENT]
+   * @param {BlendOption} [options.blendOption=BlendOption.TRANSLUCENT] Determines how primitives in the collection are blended with the scene. Must be {@link BlendOption.OPAQUE} or {@link BlendOption.TRANSLUCENT}; {@link BlendOption.OPAQUE_AND_TRANSLUCENT} is not supported.
    * @param {HeightReference} [options.heightReference=HeightReference.NONE]
    * @param {"pixels"|"meters"} [options.widthUnits="pixels"] Unit of polyline widths in this collection:
    *   <code>"pixels"</code> on the screen, or <code>"meters"</code> in world space. A clamped
    *   {@link HeightReference} measures those meters on the ellipsoid surface, so elevation and terrain
-   *   slope stretch the drawn width.
+   *   slope stretch the drawn width. Widths in meters are clamped to roughly 638 km on the ground,
+   *   beyond which a stroke's edges do not line up across tile seams.
```
Comment (mzschwartz5) [inline packages/engine/Source/Scene/BufferPolylineCollection.js]: I wouldn't be _quite_ so specific here ("638 km") -- so that we don't have to update this if we later change the clamp limit. I might rephrase this as: "Widths in meters have an upper limit to reduce discontinuities across tile seams"

**Review by mzschwartz5 — APPROVED** [review]
LGTM, nice fixes @danielzhong. Left one comment about doc phrasing, feel free to merge when ready.
